### DISCUSSION TITLE
add 2 buttons to the grid

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5460] [Help] [answered]
hi everyone 
i want to use getbutton() to add 2 buttons 
one  for excil report and the seconed one for delete the selected records

i already added it for excil as below 
getButtons() { 
            
            var buttons = super.getButtons();
            var MyRow = NomineeRow.Fields;

            buttons.push(KRVoting.Common.ExcelExportHelper.createToolButton({
                grid: this,
                onViewSubmit: () => this.onViewSubmit(),
                service: NomineeService.baseUrl + '/ListExcel',
                separator: true,
               // hint: "This is the Hint",
                title: "Nominee Report",
               
            }));

            return buttons;
        } 

so how to added another one or edit that one to delete the records
the following is the suppose code for delete 
protected getButtons() {
            return [{
                title: 'Perform Bulk Action on Selected Orders',
                cssClass: 'send-button',
                onClick: () => {
                    if (!this.onViewSubmit()) {
                        return;
                    }

                    var action = new OrderBulkAction();
                    action.done = () => this.rowSelection.resetCheckedAndRefresh();
                    action.execute(this.rowSelection.getSelectedKeys());
                }
            }];
        }

**Comment by edwardch** [comment, marked as answer]
You can just add another `buttons.push(` containing your new button code before your `return buttons;` line...

**Reply by samaremad** [reply]
thank you so much it is work correctly but i still dont know how to to delete the selected records can you help me with this ?
https://github.com/serenity-is/Serenity/discussions/5448

what suppose that i write in OrderBulkAction();

**Reply by edwardch** [reply]
Good to hear, for that take a look at the endpoint code in [BasicSamplesEndpoint.cs](https://github.com/serenity-is/Serene/blob/8f0f778d1331d3988610854e0a85e14dbff20757/Serene/Serene.Core/Modules/BasicSamples/BasicSamplesEndpoint.cs#L68).

hth
